p2p/peer.go

```diff
@@ -189,20 +266,47 @@ func (p *peer) String() string {
 // SetLogger implements BaseService.
 func (p *peer) SetLogger(l log.Logger) {
 	p.Logger = l
-	p.mconn.SetLogger(l)
 }
 
 // OnStart implements BaseService.
 func (p *peer) OnStart() error {
-	if err := p.BaseService.OnStart(); err != nil {
-		return err
+	// Open streams for all reactors.
+	p.streams = make(map[byte]transport.Stream)
+	for streamID, info := range p.streamInfoByStreamID {
+		var d transport.StreamDescriptor
+		descs := info.reactor.StreamDescriptors()
+		for _, desc := range descs {
+			if desc.StreamID() == streamID {
+				d = desc
+				break
+			}
+		}
+		stream, err := p.peerConn.OpenStream(streamID, d)
+		if err != nil {
+			return fmt.Errorf("opening stream %v: %w", streamID, err)
+		}
+		p.streams[streamID] = stream
 	}
 
-	if err := p.mconn.Start(); err != nil {
-		return err
+	// TODO: establish priority for reading from streams (consensus -> evidence -> mempool).
+	for streamID, stream := range p.streams {
+		go p.streamReadLoop(streamID, stream)
 	}
 
 	go p.metricsReporter()
+
+	// Handle connection errors.
```

Comment:
This is another go routine we add to each peer, ehehe.

Reply:
merged with metrics reporting ✅